packages/types/step/index.ts

```diff
@@ -0,0 +1,28 @@
+import {EventEmitter} from 'events';
+type StepEventHandler<T extends Step> = (e: StepEventData<T>) => any;
+type StepStatus = 'pending' | 'in progress' | 'done '; // can have more here
+export interface StepEventData<T extends Step> {
+  step: T;
+  date: Date;
+}
+interface Step extends EventEmitter {
```

Comment:
I have some ideas for processing results that will probably make this comment fade away, I'll make sure to showcase them to you later.

I'll replace nodejs' events with a cross-environment lib, [eventemitter3](https://www.npmjs.com/package/eventemitter3) looks promising and has an equivalent @types/eventemitter3